blob/drivertest/drivertest.go

```diff
@@ -42,11 +43,67 @@ type Harness interface {
 // It is called exactly once per test; Harness.Close() will be called when the test is complete.
 type HarnessMaker func(ctx context.Context, t *testing.T) (Harness, error)
 
+// AsTest represents a test of As functionality.
+// The conformance test:
+// 1. Calls BucketCheck.
+// 2. Creates a blob using Write as a WriterOption.
+// 3. Fetches the blob's attributes and calls AttributeCheck.
+// 4. Creates a Reader for the blob, and calls ReaderCheck.
+// For example, an AsTest might set a provider-specific field to a custom
+// value in Write, and then verify the custom value was returned in
+// AttributesCheck and/or ReaderCheck.
+type AsTest interface {
+	// Name should return a descriptive name for the test.
+	Name() string
+	// BucketCheck will be called to allow verification of Bucket.As.
+	BucketCheck(t *testing.T, b *blob.Bucket)
```

Comment:
I see that these methods can build up multiple failures. `Fatal` relies on the panic crossing boundaries up on the same goroutine. Instead, consider sending down an `interface{ Errorf(string, ...interface{}) }` for more harness control as the conformance tests evolve.

Reply:
Maybe these should not take a testing.T, but instead return error and let the conformance test report it?

Reply:
Done. PTAL.